synapse/lib/base.py

```diff
@@ -446,10 +446,19 @@ async def waitfini(self, timeout=None):
             return True
 
         if self.finievt is None:
```

Comment:
Why functionalize this and then not use the returned value?  Also, this pattern means we'll check for finievt being None twice?

Reply:
In order to wait on multiple things including fini, I need the event itself. Yes, this means there's an extra finievt being checked.